lib/kino.ex

```diff
@@ -193,4 +212,60 @@ defmodule Kino do
   def nothing() do
     :"do not show this result in output"
   end
+
+  @doc """
+  Starts a process under the Kino supervisor.
+
+  The process is automatically terminated when the current process
+  terminates or the current cell reevaluates.
+  """
+  @spec start_child(
+          Supervisor.child_spec()
+          | {module(), term()}
+          | module()
+        ) :: DynamicSupervisor.on_start_child()
+  def start_child(child_spec) do
+    # Starting a process that calls Kino.start_child/1 in its init
+    # would block forever, so we don't allow nesting
+    if Kino.DynamicSupervisor in Process.get(:"$ancestors", []) do
+      raise "could not start #{inspect(child_spec)} using Kino.start_child/1," <>
+              " because the current process has been started with Kino.start_child/1." <>
+              " Please move the nested start outside and pass the result as an argument to this process"
+    end
+
+    %{start: start} = child_spec = Supervisor.child_spec(child_spec, [])
+    parent = self()
+    child_spec = %{child_spec | start: {Kino, :__start_override__, [start, parent]}}
+    DynamicSupervisor.start_child(Kino.DynamicSupervisor, child_spec)
+  end
+
+  @doc false
+  def __start_override__({mod, fun, args}, parent) do
+    # We switch the group leader, so that the newly started
+    # process gets the same group leder as the caller
+    initial_gl = Process.group_leader()
+    gl = Process.info(parent)[:group_leader]
+
+    Process.group_leader(self(), gl)
+
+    try do
+      apply(mod, fun, args)
+      |> case do
+        {:ok, pid} = resp -> {:ok, resp, pid}
+        {:ok, pid, _info} = resp -> {:ok, resp, pid}
+        resp -> {:error, resp}
+      end
+      |> case do
+        {:ok, resp, pid} ->
+          Kino.Bridge.object_add_pointer(pid, parent)
+          Kino.Bridge.object_monitor(pid, Kino.Terminator.cross_node_name(), {:terminate, pid})
```

Comment:
Can we do this operation in a single pass?

Have object_monitor receive pid, parent, target and message. If there is no pointer, then you add one behind the scenes. The goal is to reduce the communication surface with Livebook.

```suggestion
          Kino.Bridge.object_add_pointer(pid, parent)
          Kino.Bridge.object_monitor(pid, Kino.Terminator.cross_node_name(), {:terminate, pid})
```

Reply:
> Can we do this operation in a single pass?

Sure, but then we will essentially have `object_add_pointer_and_monitor` + `object_add_pointer` that we need in other places. So we would optimise this case into a single message, but `object_monitor` is really a separate action that we could use for something else in the future.

> The goal is to reduce the communication surface with Livebook.

The surface doesn't change though, since we still need at least two actions?

> Also, we only monitor PIDs right? So maybe we should remove object_ from the naming.

This can be anything, in particular we monitor control/input refs to notify SubscriptionManager once it disappears.